R/convert_docx.R

```diff
@@ -0,0 +1,74 @@
+#' @title Convert a docx-file (a protocol) to an (R)markdown file
+#'
+#' @description This function is derived from the \link[redoc]{dedoc} and uses pandoc to convert between docx and markdown. Several options are preset to end-up with a markdown document that is in syntax as close as possible to Rmarkdown files in RStudio. During conversion, graphics (e.g. png, jpg) will be extracted from the docx archive and placed in a folder `./media` and named `image1`, `image2`, etcetera.
+#'
+#' @details Metadata in the page headers and footers of the docx are ignored and will thus be lost during conversion. In case the header or footer did contain important metadata, it will need to be recovered manually. Usually header information will go inside a yaml section of an Rmarkdown.
+#'
+#' @param docx The `.docx` file to convert. Can be given as an absolute or relative path.
+#' @param to The filename to write the resulting `.Rmd` file (without path). The default is to
+#' use the same basename as the docx document.
+#' @param dir The directory to write the `.Rmd` to. Defaults to current working
+#' directory. Any images will be written to `dir/media`.
+#' @param wrap The width at which to wrap text. If `NA`, text is not wrapped.
+#' Defaults to 80.
+#' @param overwrite Whether or not to overwrite the to file if it already
+#' existed. Defaults to FALSE.
+#' @param verbose whether to print pandoc progress text. Defaults to FALSE.
+#'
+#' @importFrom rmarkdown pandoc_convert
+#' @importFrom tools file_path_sans_ext
+#' @importFrom stringr str_replace_all
+#' @export
+docx_to_md <- function(
+  docx,
+  to = NULL,
+  dir = ".",
+  wrap = 80,
+  overwrite = FALSE,
+  verbose = FALSE) {
+
+  if (is.null(to)) to <- paste0(file_path_sans_ext(basename(docx)), ".Rmd")
+  if (is.null(dir)) dir <- "."
+  to <- file.path(dir, to)
+  if (!overwrite && file.exists(to)) stop(to, " exists and overwrite = FALSE")
+
+
+  md <- convert_docx_to_md(docx, wrap, verbose)
+  md <- str_replace_all(md, pattern = "\\r", replacement = "")
+
+
+  cat(md, file = to, sep = "")
+  return(to)
+}
+
+#' @importFrom rmarkdown pandoc_convert
+convert_docx_to_md <- function(docx,
+                               wrap,
+                               verbose) {
+  docx <- normalizePath(docx)
+
+  if (is.null(wrap)) {
```

Comment:
```suggestion
  if (missing(wrap)) {
```

I recommend to check `wrap` in case it is given by the user